File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> ans = new ArrayList<>();

        Queue<TreeNode> queue = new LinkedList<>();

        if(root == null){
            return ans;
        }

        queue.add(root);
        ans.add(new ArrayList<>(Arrays.asList(root.val)));
        int len;

        while(!queue.isEmpty()){
            len = queue.size();
            List<Integer> currLevel = new ArrayList<>();


            while(len>0){
                TreeNode currNode = queue.remove();
                len--;

                if(currNode.left != null){
                    queue.add(currNode.left);
                    currLevel.add(currNode.left.val);
                }
                if(currNode.right != null){
                    queue.add(currNode.right);
                    currLevel.add(currNode.right.val);
                }

            }

            if(currLevel.size()>0){
                ans.add(currLevel);
            }
        }


        return ans;
    }
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        """
        lever order traversal?
        at each level pick the last element

        """
        if not root:
            return []
        res = []
        q = deque()

        q.append((root,0))
        levelMap = defaultdict(int)
        levelMap[0] = 1


        while(len(q) != 0):
            r, lvl = q.popleft()
            if levelMap[lvl] == 1:
                res.append(r.val)
            levelMap[lvl] -= 1


            if r.left is not None:
                q.append((r.left,lvl+1))
                levelMap[lvl+1] += 1
            if r.right is not None:
                q.append((r.right,lvl+1))
                levelMap[lvl+1] += 1  
        return res
            


        